### ISSUE TITLE
The bundle at path Payload/MyApp.app/Frameworks/_CollectionsUtilities.framework has an invalid CFBundleIdentifier '_CollectionsUtilities'

### ISSUE THREAD

**Issue by kimdv** [apple/swift-collections#364]
### Information

- **Package version:** 1.1.0
- **Platform version:** [Please tell us the version number of your operating system.](https://github.com/actions/runner-images/blob/7535fb59ffbdb1638ace2d693ccc244fb2faf41c/images/macos/macos-13-Readme.md)
- **Swift version:** 5.10

### Checklist

- [ ] If possible, I've reproduced the issue using the `main` branch of this package.
- [x] I've searched for [existing GitHub issues](https://github.com/apple/swift-collections/issues).

### Steps to Reproduce
Replace this paragraph with an explanation of how to reproduce the incorrect behavior.
Include a simple code example, if possible.

### Expected behavior
It should compile fine allow to be uploaded to AppStore

### Actual behavior
When using 1.1.0 to upload our app to the AppStore we get the following error:

```
ERROR [2024-02-22 20:51:10.35]: ERROR: [ContentDelivery.Uploader] Asset validation failed (90049) This bundle is invalid. The bundle at path Payload/MyApp.app/Frameworks/_CollectionsUtilities.framework has an invalid CFBundleIdentifier '_CollectionsUtilities' There are invalid characters(characters that are not dots, hyphen and alphanumerics) that have been replaced with their code point '\u005fCollectionsUtilities' CFBundleIdentifier must be present, must contain only alphanumerics, dots, hyphens and must not end with a dot. [see the Core Foundation Keys at https://developer.apple.com/library/content/documentation/General/Reference/InfoPlistKeyReference/Articles/CoreFoundationKeys.html#//apple_ref/doc/uid/TP40009249-102070-TPXREF105] (ID: d3c9cf5e-809e-40ad-b851-c1acd4ee9056)
```

The app compiles fine

**Comment by kimdv** [comment]
Some more info: 

```
2024-02-22T19:51:05.1131850Z ▸ Processing empty-_CollectionsUtilities.plist
2024-02-22T19:51:05.1154810Z     invalid character in Bundle Identifier. This string must be a uniform type identifier (UTI) that contains only alphanumeric (A-Z,a-z,0-9), hyphen (-), and period (.) characters. (in target '_CollectionsUtilities' from project 'swift-collections')
```

**Comment by lorentey** [comment]
That's not ideal.

- What platform are you targeting?
- Can you step me through your build setup? In particular, did you do any custom configuration to have this package be built as a set of standalone frameworks, as opposed to directly linking it into the app executable?

**Comment by kimdv** [comment]
> That's not ideal.
> 
> * What platform are you targeting?

We have iOS 14 ar min target version.

> * Can you step me through your build setup? In particular, did you do any custom configuration to have this package be built as a set of standalone frameworks, as opposed to directly linking it into the app executable?

![Screenshot 2024-02-28 at 12 46 53](https://github.com/apple/swift-collections/assets/5477711/6f54efbc-3421-4fef-b78c-b7aafe0ad66f)

We have added it to our Xcode project under the "Package Dependencies" tab.
In our app target we import "Collections" and in a framework target we import "OrderedCollections". 
![Screenshot 2024-02-28 at 12 50 39](https://github.com/apple/swift-collections/assets/5477711/186e6d48-f45c-4f10-b52b-7c19dfe56108)
![Screenshot 2024-02-28 at 12 50 50](https://github.com/apple/swift-collections/assets/5477711/b368de91-da54-4047-9c37-cefd9791e665)



Does that help?

**Comment by depth42** [comment]
I am running into the same problem. The culprit seems to be the underscore in the CFBundleIdentifier _CollectionUtilities which is put into the generated Info.plist. Underscores are forbidden in these identifiers. This even prevents uploading apps using this to App Store Connect. An upload results in an error.

**Comment by lorentey** [comment]
@kimdv Yep, that's quite clear.

SwiftPM's Xcode integration should not ever generate invalid bundle identifiers -- so the cause is clearly a tooling bug. However, the only way we can resolve this without waiting for a tools update is to rename the module. That takes a little bit of coordination, but it's a straightforward change and it can ship in 1.1.1.

**Comment by vanvoorden** [comment]
> We have added it to our Xcode project under the "Package Dependencies" tab.

@kimdv Hmm… is there any utility or tool I could run locally to confirm the bug in a demo app? Have you experimented with any workarounds so far?

> In our app target we import "Collections" and in a framework target we import "OrderedCollections".

Did you experiment with depending on the entire `Collections` package in the framework target and then just having your app target depend on the framework (which exposes `Collections`)? Same problem?

**Comment by lorentey** [comment]
I see a warning when I try reproducing this:

```
Module 'HeapModule' was not compiled with library evolution support; using it means binary compatibility for 'TestFramework' can't be guaranteed
```

@kimdv do you see this too? Are you doing anything to suppress this warning?

**Comment by depth42** [comment]
I am seeing the same warning when doing a debug build. When uploading to App Store Connect, this turns into an error.

**Comment by kimdv** [comment]
> @kimdv Hmm… is there any utility or tool I could run locally to confirm the bug in a demo app? Have you experimented with any workarounds so far?

We had one place where we imported Collections and one place where use Collections and one place OrderedCollections. 
You need to upload an app to the AppStore to get this error.
 
> > In our app target we import "Collections" and in a framework target we import "OrderedCollections".
> 
> Did you experiment with depending on the entire `Collections` package in the framework target and then just having your app target depend on the framework (which exposes `Collections`)? Same problem?

We have one target using Collections, and I will assume it's the same as it imports OrderedCollections.

**Comment by ptliddle** [comment]
I've also just hit this problem trying to release to the AppStore. Is there any resolution or work around at the moment?

**Comment by depth42** [comment]
I‘ve forked the repository and removed the underscore. This got ridnof the warning / error.
https://github.com/projectwizards/swift-collections

**Comment by jtaby** [comment]
Wanted to add that we're encountering this same issue

**Comment by lorentey** [comment]
#382 has landed on release/1.1; it works around this issue while we are waiting for an Xcode-level fix.

This work is scheduled to be part of the upcoming 1.1.1 release.
